.ado/publish.yml

```diff
@@ -16,398 +73,408 @@ variables:
+          # and we can't run the samples as qsc is currently cross compiled.
+          - script: |
+              env CARGO_BUILD_TARGET=aarch64-unknown-linux-gnu python build.py --pip --no-check-prereqs --no-check --no-test
+            displayName: Cross Build Linux aarch64 Py Packages
+            condition: and(eq(variables['Agent.OS'], 'Linux'), eq(variables['arch'], 'aarch64'))
+
+          - script: |
+              chmod +x ./docker/linux-aarch64/build.sh
+              chmod +x ./docker/linux-aarch64/run.sh
+
+              ./docker/linux-aarch64/build.sh
+              ./docker/linux-aarch64/run.sh
+            displayName: Run auditwheel and python tests for Linux aarch64 Wheels
+            condition: and(eq(variables['Agent.OS'], 'Linux'), eq(variables['arch'], 'aarch64'))
+
+          - script: |
+              echo ##vso[task.setvariable variable=CARGO_BUILD_TARGET]aarch64-pc-windows-msvc
+            displayName: Set cargo build target for Windows aarch64
+            condition: and(eq(variables['Agent.OS'], 'Windows_NT'), eq(variables['arch'], 'aarch64'))
+
+          - script: |
+              python build.py --pip --no-check-prereqs --no-check --no-test
+            displayName: Cross Build Windows aarch64 Py Packages
+            condition: and(eq(variables['Agent.OS'], 'Windows_NT'), eq(variables['arch'], 'aarch64'))
+
+          - script: |
+              dir target\wheels\*
+            displayName: List Py Packages on Win
+            condition: eq(variables['Agent.OS'], 'Windows_NT')
+
+          - script: |
+              ls target/wheels/*
+            displayName: List Py Packages on non-Win
+            condition: ne(variables['Agent.OS'], 'Windows_NT')
+
+    - stage: approval
+      displayName: Approval
+      dependsOn: build
+      condition: eq(variables['Build.Reason'], 'Manual')
+      jobs:
+      - job: "Approval"
+        pool: server
+        timeoutInMinutes: 1440 # job times out in 1 day
+        steps:
+        - task: ManualValidation@0
+          timeoutInMinutes: 1440 # task times out in 1 day
+          inputs:
+            notifyUsers: ''
+            instructions: 'Please verify artifacts and approve the release'
+            onTimeout: 'reject'
+
+    - stage: release
+      displayName: Release
+      dependsOn: approval
+      condition: eq(variables['Build.Reason'], 'Manual')
+      jobs:
+      - job: "Publish_VSIX_Package"
+        pool:
+          name: 'Azure-Pipelines-DevTools-EO'
+          image: 'ubuntu-latest'
+          os: linux
+        templateContext:
+          type: releaseJob
+          isProduction: true
+          inputs:  # All input build artifacts must be declared here
+          - input: pipelineArtifact
+            artifactName: VSIX
+            targetPath: $(System.DefaultWorkingDirectory)/target/vscode
+        steps:
+        - script: |
+            npm install -g @vscode/vsce
```

Comment:
Sounds silly, but I suggest moving this to pre-approval if possible. I think we should have as few steps for actual publishing as possible so that we have fewer failures after approval. I don't see why this should fail, but it would be beneficial to catch failures in the part that executes often.
(Feel free to ignore.)

Reply:
This can't be moved as we are on a fresh image for the job and no data from the previous jobs is available except that which is fed as a pipeline artifact.